ingestion-beam/src/main/java/com/mozilla/telemetry/util/Normalize.java

```diff
@@ -4,28 +4,54 @@
 
 package com.mozilla.telemetry.util;
 
+import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableList;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
 
 public class Normalize {
 
+  private static final String OTHER = "Other";
+
   private static final String BETA = "beta";
   private static final String NIGHTLY = "nightly";
-  private static final String OTHER = "Other";
   private static final List<String> CHANNELS = ImmutableList.of("release", "esr", BETA, "aurora",
       NIGHTLY);
 
+  private static final String WINDOWS = "Windows";
+  private static final String MAC = "Mac";
+  private static final String LINUX = "Linux";
+  private static final String IOS = "iOS";
+  private static final String ANDROID = "Android";
+
+  private static final List<String> APP_NAMES = ImmutableList.of(
+      // Make sure more-specific versions come before Zerda.
+      "Zerda_cn", "Zerda",
```

Comment:
I think there's a bug in [the existing lua implementation of app name normalization](https://github.com/mozilla-services/lua_sandbox_extensions/blob/607b3c7b9def600ddbd373a14f67a55e1c69f7a1/moz_telemetry/modules/moz_telemetry/normalize.lua#L218-L230) where Zerda is checked before Zerda_cn, so "Zerda_cn" will be normalized to "Zerda". Same for "Focus-TV" normalized to "Focus".

Reply:
Yeah, they should be reordered
https://github.com/mozilla-services/lua_sandbox_extensions/pull/431